<launch>
   <rosparam command="load" file="$(find collision_avoidance)/config/collision_avoidance.yaml"/>
  <arg name="enableRviz" default="true"/>
  <arg name="mrs" default="true"/>
  <arg name="cola2_architecture" default="true"/>
  <arg name="number_of_robots" default="2"/> 

  <!-- COLA2 architecture -->
  <group if="$(arg cola2_architecture)">
    <!-- Static transform -->
    <node name="world2ned" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 3.1415 world world_ned 100"/>

    <!-- Include AUVs -->
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 1)">
      <arg name="robot_name" value="sparus_0"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 2)">
      <arg name="robot_name" value="sparus_1"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 3)">
      <arg name="robot_name" value="sparus_2"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 4)">
      <arg name="robot_name" value="sparus_3"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 5)">
      <arg name="robot_name" value="sparus_4"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 6)">
      <arg name="robot_name" value="sparus_5"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 7)">
      <arg name="robot_name" value="sparus_6"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 8)">
      <arg name="robot_name" value="sparus_7"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 9)">
      <arg name="robot_name" value="sparus_8"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 10)">
      <arg name="robot_name" value="sparus_9"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    <include file="$(find cola2_sparus2)/launch/sim_start.launch" if="$(eval int(arg('number_of_robots')) >= 11)">
      <arg name="robot_name" value="sparus_10"/>
      <arg name="enable_keyboard" value="true"/>
    </include>
    
  </group>

  <!-- Multi-robot system -->
  <group if="$(arg mrs)">
    <group ns="/mrs">
      <!-- Load configuration files -->
      <rosparam command="load" file="$(find collision_avoidance)/config/collision_avoidance.yaml"/>

      <!-- Dynamically add AUV nodes -->
      <node name="AUV_sparus_0" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 1)">
        <param name="robot_ID" value="0"/>
        <param name="robot_name" value="sparus_0"/>
        <param name="navigation_depth" value="0"/>
      </node>
      <node name="AUV_sparus_1" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 2)">
        <param name="robot_ID" value="1"/>
        <param name="robot_name" value="sparus_1"/>
        <param name="navigation_depth" value="0"/>
      </node>
      <node name="AUV_sparus_2" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 3)">
        <param name="robot_ID" value="2"/>
        <param name="robot_name" value="sparus_2"/>
        <param name="navigation_depth" value="0"/>
      </node>
      <node name="AUV_sparus_3" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 4)">
        <param name="robot_ID" value="3"/>
        <param name="robot_name" value="sparus_3"/>
        <param name="navigation_depth" value="0"/>
      </node>
      <node name="AUV_sparus_4" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 5)">
        <param name="robot_ID" value="4"/>
        <param name="robot_name" value="sparus_4"/>
        <param name="navigation_depth" value="0"/>
      </node>
      <node name="AUV_sparus_5" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 6)">
        <param name="robot_ID" value="5"/>
        <param name="robot_name" value="sparus_5"/>
        <param name="navigation_depth" value="0"/>
      </node>
      <node name="AUV_sparus_6" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 7)">
        <param name="robot_ID" value="6"/>
        <param name="robot_name" value="sparus_6"/>
        <param name="navigation_depth" value="0"/>
      </node>
      <node name="AUV_sparus_7" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 8)">
        <param name="robot_ID" value="7"/>
        <param name="robot_name" value="sparus_7"/>
        <param name="navigation_depth" value="0"/>
      </node>
      <node name="AUV_sparus_8" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 9)">
        <param name="robot_ID" value="8"/>
        <param name="robot_name" value="sparus_8"/>
        <param name="navigation_depth" value="0"/>
      </node>
      <node name="AUV_sparus_9" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 10)">
        <param name="robot_ID" value="9"/>
        <param name="robot_name" value="sparus_9"/>
        <param name="navigation_depth" value="0"/>
      </node>
      <node name="AUV_sparus_10" pkg="collision_avoidance" type="AUV_robot.py" output="screen" respawn="true" if="$(eval int(arg('number_of_robots')) >= 11)">
        <param name="robot_ID" value="10"/>
        <param name="robot_name" value="sparus_10"/>
        <param name="navigation_depth" value="0"/>
      </node>
    </group>
  </group>
  <!-- RViz -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find collision_avoidance)/scene/sparus2.rviz"/>
</launch>
